### PR TITLE
role grants + admin

### PR DESCRIPTION
**Pull request by UnsungHero97**

### CODE REVIEW

File: src/containers/orgs/OrgsReducer.js [hunk 1]
```diff
@@ -155,6 +168,31 @@ export default function orgsReducer(state :Map = INITIAL_STATE, action :Object)
...
+          .setIn([ADD_CONNECTION, 'requestState'], RequestStates.PENDING)
+          .setIn([ADD_CONNECTION, seqAction.id], seqAction),
+        SUCCESS: () => {
+          const storedSeqAction :SequenceAction = state.getIn([ADD_CONNECTION, seqAction.id]);
+          if (storedSeqAction) {
+            const { connection, organizationId } = storedSeqAction.value;
+            return state
+              .setIn([ADD_CONNECTION, 'requestState'], RequestStates.SUCCESS)
+              .updateIn(
+                ['orgs', organizationId, 'connections'],
```
Comment (bergieboy) [inline src/containers/orgs/OrgsReducer.js]: Should `requestState`, `orgs`, and `connections` be consts?

File: src/containers/orgs/OrgsReducer.js [hunk 2]
```diff
@@ -710,6 +773,29 @@ export default function orgsReducer(state :Map = INITIAL_STATE, action :Object)
...
+          .setIn([UPDATE_ROLE_GRANT, 'requestState'], RequestStates.PENDING)
+          .setIn([UPDATE_ROLE_GRANT, seqAction.id], seqAction),
+        SUCCESS: () => {
+          const storedSeqAction :SequenceAction = state.getIn([UPDATE_ROLE_GRANT, seqAction.id]);
+          if (storedSeqAction) {
+            const grant :Grant = storedSeqAction.value.grant;
+            const organizationId :UUID = storedSeqAction.value.organizationId;
+            const roleId :UUID = storedSeqAction.value.roleId;
+            return state
+              .setIn(['orgs', organizationId, 'grants', roleId], grant.toImmutable())
```
Comment (bergieboy) [inline src/containers/orgs/OrgsReducer.js]: should `grants` be a const?

File: src/containers/orgs/OrgsSagas.js [hunk 3]
```diff
@@ -53,21 +58,67 @@ const { PermissionTypes, PrincipalTypes } = Types;
...
   getOrganizationIntegrationAccount,
   getOrganizationMembers,
+  removeConnections,
 } = OrganizationsApiActions;
 const {
+  addConnectionsWorker,
   getAllOrganizationsWorker,
   getOrganizationWorker,
   getOrganizationIntegrationAccountWorker,
+  removeConnectionsWorker,
```
Comment (bergieboy) [inline src/containers/orgs/OrgsSagas.js]: What are these foreign beasts?

File: src/containers/orgs/components/OrgConnectionsSection.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        </div>
+      </SectionGrid>
+    );
+  }
+}
+
+const mapStateToProps = (state :Map) => ({
+  requestStates: {
+    [ADD_CONNECTION]: state.getIn(['orgs', ADD_CONNECTION, 'requestState']),
+    [REMOVE_CONNECTION]: state.getIn(['orgs', REMOVE_CONNECTION, 'requestState']),
```
Comment (bergieboy) [inline src/containers/orgs/components/OrgConnectionsSection.js]: I love this

File: src/containers/orgs/OrgsSagas.js (hunk 3 above)
Comment (UnsungHero97) [reply src/containers/orgs/OrgsSagas.js]: Whaaat?

File: src/containers/orgs/OrgsReducer.js (hunk 1 above)
Comment (UnsungHero97) [reply src/containers/orgs/OrgsReducer.js]: I know you all do something different in other apps but I don't know how I feel about it. 

I agree. I don't like having these magic strings all over the place. However, I don't feel like this is any better:

```
import { REDUX_CONSTANTS } from '../../core/redux';
const { ORGS, CONNECTIONS } = REDUX_CONSTANTS;
...
const mapStateToProps = (state :Map) => ({
  requestStates: {
    [UPDATE_ROLE_GRANT]: state.getIn([ORGS, UPDATE_ROLE_GRANT, REQUEST_STATE]),
  }
});
```

Comment (bergieboy) [reply src/containers/orgs/OrgsReducer.js]: Yeah, I think the the benefit is that if you need to change that string for whatever reason, you can do it in one place. You could argue that that the const name could just as easily be misspelled as the string, but the at least linter would yell at you for that. You do you though! Just a suggestion.

Comment (UnsungHero97) [reply src/containers/orgs/OrgsReducer.js]: Solid points. I'll start playing with this approach going forward to see how it feels.

**Pull request merged**